### ISSUE TITLE
Fix mac builds

### ISSUE THREAD

**Issue by freaktechnik** [nightingale-media-player/nightingale-hacking#354]
Apparently some (or all?) mac builds are broken: http://forum.getnightingale.com/thread-429-post-5394.html#pid5394

According to @zjays it's 3267acc1883a3d33eff876c8bdca4b961c29fdeb that broke it.

Their configure log looks like this:

```
`Adding configure options from /Users/Zach-Mac/GitHub/nightingale-hacking/nightingale.config:
  --with-macosx-sdk=/Developer/SDKs/MacOSX10.5.sdk
  --enable-installer
  --enable-official
  --enable-compiler-environment-checks=no
rm -f /Users/Zach-Mac/GitHub/nightingale-hacking/configure /Users/Zach-Mac/GitHub/nightingale-hacking/.songbirdconfig.mk /Users/Zach-Mac/GitHub/nightingale-hacking/.songbirdconfig.out 
rm -f -r /Users/Zach-Mac/GitHub/nightingale-hacking/compiled
Adding configure options from /Users/Zach-Mac/GitHub/nightingale-hacking/nightingale.config:
  --with-macosx-sdk=/Developer/SDKs/MacOSX10.5.sdk
  --enable-installer
  --enable-official
  --enable-compiler-environment-checks=no
Nightingale Build System
cd /Users/Zach-Mac/GitHub/nightingale-hacking && \
    autoconf && \
    rm -f -r /Users/Zach-Mac/GitHub/nightingale-hacking/autom4te.cache/
mkdir -p /Users/Zach-Mac/GitHub/nightingale-hacking/compiled /Users/Zach-Mac/GitHub/nightingale-hacking/compiled/dist
cd /Users/Zach-Mac/GitHub/nightingale-hacking/compiled && \
    /Users/Zach-Mac/GitHub/nightingale-hacking/configure --with-macosx-sdk=/Developer/SDKs/MacOSX10.5.sdk --enable-installer --enable-official --enable-compiler-environment-checks=no 
checking build system type... x86_64-apple-darwin12.6.0
checking host system type... x86_64-apple-darwin12.6.0
checking target system type... x86_64-apple-darwin12.6.0
checking for git... git
configure: error: Unsupported architecture
make: *** [/Users/Zach-Mac/GitHub/nightingale-hacking/compiled/config.status] Error 1
```

Apparently setting `--disable-compiler-environment-checks` doesn't help either.

From the changes in the configure update, it now detects the architecture on mac (from [line 1150 in the old version](https://github.com/nightingale-media-player/nightingale-hacking/commit/3267acc1883a3d33eff876c8bdca4b961c29fdeb#diff-93452acb07da1f0393119ce86e69e3bbL1150)) , and I'm not sure how that works for nightingale.

**Comment by freaktechnik** [comment]
I guess https://github.com/nightingale-media-player/nightingale-hacking/blob/af788cb0ce92881108634828f80361341a3113f6/build/config.mk would have to be updated.

**Comment by darealshinji** [comment]
Does it fix the Mac builds if config.sub and config.guess are replaced with the most recent versions?
http://git.savannah.gnu.org/gitweb/?p=config.git;a=blob_plain;f=config.guess;hb=HEAD
http://git.savannah.gnu.org/gitweb/?p=config.git;a=blob_plain;f=config.sub;hb=HEAD

Otherwise I guess they should be restored to the old versions:
https://raw.githubusercontent.com/nightingale-media-player/nightingale-hacking/7af3844ae143c05f7ac8a83805a40e7abb4402bc/build/autoconf/config.guess
https://raw.githubusercontent.com/nightingale-media-player/nightingale-hacking/7af3844ae143c05f7ac8a83805a40e7abb4402bc/build/autoconf/config.sub

**Comment by zjays** [comment]
@darealshinji I tried the most recent versions of config.sub and config.guess, but got the same error as before.

Here is a portion of my configure log when building before 3267acc - it thinks I'm on a 32-bit system:

```
checking build system type... i686-apple-darwin12.6.0
checking host system type... i686-apple-darwin12.6.0
checking target system type... i686-apple-darwin12.6.0
configure: WARNING: Compiler environment checks disabled.
```

So the updated config.sub and/or config.guess correctly detects that my system is 64-bit. Turns out there was no entry for 64-bit macs in configure.ac, so I added it: 7c5e2cd

After that, the building tried to look for dependencies in the non-existent macosx-x86_64 dep folder.

```
make -C /Users/Zach-Mac/github/nightingale-hacking/compiled
/Users/Zach-Mac/github/nightingale-hacking/build/config.mk:44: *** Please check out or build vendor-binaries - /Users/Zach-Mac/github/nightingale-hacking/dependencies/macosx-x86_64/mozilla-1.9.2/release/scripts see http://wiki.getnightingale.com/doku.php?id=build#build_it_yourself.  Stop.
make: *** [build] Error 2
```

So as a temporary solution I made a copy of the macosx-i686 folder, renamed it macosx-x86_64 and placed it in the dependencies folder. After doing this ./build.sh reported Success, but the compiled Mac package was missing several .dylib files, causing it to crash upon starting:

```
Dyld Error Message:
image not found /Users/Zach-Mac/GitHub/nightingale-hacking/compiled/dist/Nightingale.app /Contents/Resources/lib/libintl.dylib
```

Full terminal output posted here: http://forum.getnightingale.com/thread-429-post-5482.html#pid5482

So to sum up, something messes up the build process if the mac system is reported as x86_64. Rather than reverting 3267acc, I propose just making a small change to config.guess to keep reporting 64-bit macs as i686 (like so: 76de53d). This fixed mac building for me...but then it broke again after the verbosity changes were merged in (see my comment on https://github.com/nightingale-media-player/nightingale-hacking/pull/336).

**Comment by darealshinji** [comment]
I tried to fix the Makefile verbosity stuff. I hope it now works.

**Comment by freaktechnik** [comment]
The verbostiy stuff was fixed in c016b25bb5d265de75827782131969366d7513d2
